### PR TITLE
Improve usability

### PR DESCRIPTION
**Pull request by BastianStender**
Miscellaneous changes to improve overall usability:
- do not run tests if init failed
- raise additional error on failed serial open
- show steps in GraphStrategy
- use check_output instead of check_call for more informative exceptions and debug log command as well as output

### CODE REVIEW

File: labgrid/pytestplugin/fixtures.py [hunk 1]
```diff
@@ -113,5 +114,9 @@ def env(request):
 def target(env):
     """Return the default target `main` configured in the supplied
     configuration file."""
-    target = env.get_target()
+    try:
+        target = env.get_target()
+    except UserError as e:
```
Comment (jluebbe) [inline labgrid/pytestplugin/fixtures.py]: Why do you limit this to UserErrors? We probably can't recover on other errors as well.

File: labgrid/driver/fastbootdriver.py [hunk 2]
```diff
@@ -43,7 +45,10 @@ def on_deactivate(self):
     @Driver.check_active
     @step(title='call', args=['args'])
     def __call__(self, *args):
-        subprocess.check_call(self._get_fastboot_prefix() + list(args))
+        cmd = self._get_fastboot_prefix() + list(args)
+        self.logger.debug(' '.join(cmd))
+        stdout = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+        self.logger.debug(stdout)
```
Comment (jluebbe) [inline labgrid/driver/fastbootdriver.py]: Hmm. Currently, if fastboot fails and you're not running in pytest with stdout capture, you just have the error message from fastboot on the console. That's gone with your change. Do we want to pass the output to a higher log level if an exception happens?

File: labgrid/pytestplugin/fixtures.py (hunk 1 above)
Comment (BastianStender) [reply labgrid/pytestplugin/fixtures.py]: Yes, that is true. Do we want to also exit on `target is None`? What's the reasoning behind `get_target()` returning `None` anyway? Shouldn't this be an exception instead? Or is returning `None` the workaround to handle a unsuccessful `get_target()` during test runtime?

File: labgrid/driver/fastbootdriver.py (hunk 2 above)
Comment (BastianStender) [reply labgrid/driver/fastbootdriver.py]: Why is stdout output gone with my change? The exception should contain that, see my commit message.

**Comment by BastianStender** [comment]
Pushed changes:
- add strategy/graphstrategy: do not allow unknown states in via
- add strategy/graphstrategy: raise exception on impossible paths with via

File: labgrid/pytestplugin/fixtures.py (hunk 1 above)
Comment (jluebbe) [reply labgrid/pytestplugin/fixtures.py]: get_target() returning None can also be treated as fatal here, but with an explicit error message. Using pytest.exit here should be ok for now, as this fixture is used very early.

**Comment by BastianStender** [comment]
Pushed changes:
- drop check_output() changes
- pytestplugin/fixtures: raise UserError on `target is None` in target fixture

**Comment by BastianStender** [comment]
Pushed changes:
- rebased on current master

**Pull request merged**